Ncsdk 34445 ble samples test applications (#23400)
nrfconnect/sdk-nrf: tests/samples/bluetooth/samples_test_app/README.rst

.. _samples_test_app:

> Bluetooth: BLE Master Test App

Author: It is an app, it is in tests/samples folder, as it is used for testing samples

.. _samples_test_app:

Bluetooth: Bluetooth LE Master Test app
#######################################

.. contents::
   :local:
   :depth: 2

The Bluetooth LE Master Test app is a Bluetooth® Low Energy (LE) central application designed to test basic Bluetooth functionality in peripheral samples.
This application acts as a tester in the central role to validate Bluetooth LE peripheral applications in the |NCS|.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

.. include:: /includes/tfm.txt

The sample requires a Bluetooth LE peripheral device for testing.
You can use any of the following peripheral samples:

* :ref:`peripheral_uart`
* :ref:`peripheral_hids_keyboard`
* :ref:`peripheral_hids_mouse`
* :ref:`peripheral_lbs`

Overview
********

> The Bluetooth LE Master Test app implements a BLE central device that can discover and connect to peripheral devices.

Reviewer: ```suggestion
The Bluetooth LE Master Test app implements a Bluetooth LE central device that can discover and connect to peripheral devices.
```

It performs automatic device discovery, service discovery, connection management, and supports security features.

The application follows a specific test sequence:

> 1. Initialization - Bluetooth stack initialization and module setup
> #. Scanning - Active scanning for supported peripheral devices
> #. Connection - Automatic connection to discovered devices
> #. Service discovery - GATT service discovery and logging
> #. Connection maintenance - Maintains connection for 5 seconds
> #. Disconnection - Graceful disconnection
> #. Reconnection - Waits 3 seconds, then reconnects
> #. Extended test - Maintains connection for 10 seconds
> #. Final disconnection - Completes the test cycle
> #. Success indication - Prints ``SUCCESS`` when all steps complete

Reviewer: Rephrased for consistency .
```suggestion
1. Initialization - Initializes the Bluetooth stack and sets up required modules.
#. Scanning - Actively scans for supported peripheral devices.
#. Connection - Automatically connects to discovered devices.
#. Service discovery - Discovers and logs GATT services.
#. Connection maintenance - Maintains the connection for 5 seconds.
#. Disconnection - Disconnects gracefully.
#. Reconnection - Waits 3 seconds, then reconnects.
#. Extended test - Maintains the connection for 10 seconds.
#. Final disconnection - Completes the test cycle.
#. Success indication - Prints ``SUCCESS`` when all steps are completed.
```

Supported device types
======================

The application is configured to discover and test the following Bluetooth LE peripheral devices:

* Nordic UART Service
* Nordic Throughput
* NCS HIDS Keyboard
* NCS HIDS Mouse
* Nordic Lightweight Battery Service (LBS)
* Test Beacon
* HID devices (through UUID filtering)

Debugging
*********

In this sample, the UART console is used to display test progress and results.
Debug messages are not displayed in the UART console, instead, they are printed by the RTT logger.

> If you want to view the debug messages, follow the procedure described in :ref:`testing_rtt_connect`.

Reviewer: ```suggestion
To view debug messages, follow the procedure described in :ref:`testing_rtt_connect`.
```

      #. |connect_terminal|
      #. Reset the kit.
      #. Observe that the text ``Starting Bluetooth LE test application`` is printed on the COM listener.
      #. Power on a Bluetooth LE peripheral device that matches one of the supported device types.
      #. Observe that the application discovers and connects to the peripheral device.
      #. Monitor the console output for successful connection and service discovery.
      #. Verify that the application completes the full test cycle and prints ``SUCCESS``.

Expected behavior
=================

The application provides detailed console output, including:

* Bluetooth initialization status
* Device discovery information
* Connection establishment details
* Service discovery results
* Security pairing information
* Test progress and results

The following is an example of the output:

.. code-block:: console

   Starting BLE test application
   Bluetooth initialized
   UART initialized
   Scan module initialized
   Scanning successfully started
   Filters matched. Address: AA:BB:CC:DD:EE:FF connectable: 1
   Connected
   Service discovery completed
   Disconnect successful
   SUCCESS

Testing workflow
================

> 

Reviewer: The application supports both manual and automated testing.

Manual testing
--------------

To test the sample manually, complete the following steps:

1. Build and flash the application to a development kit.
#. Power on a Bluetooth LE peripheral device (for example, Nordic UART Service sample).
#. Monitor the console output for successful connection and service discovery.
#. Verify that the application completes the full test cycle.

Automated testing
-----------------

> The application is designed for integration with automated test frameworks, offering:

Reviewer: ```suggestion
The application is designed for integration with automated test frameworks, providing:
```

.. _samples_test_app:

Bluetooth: BLE Master Test App
##############################

.. contents::
   :local:
   :depth: 2

The BLE Master Test App is a Bluetooth Low Energy (BLE) central application designed for testing basic Bluetooth functionality in peripheral samples. This application acts as a tester with central role to validate BLE peripheral applications in the Nordic Connect SDK (NCS).

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

.. include:: /includes/tfm.txt

The sample also requires a BLE peripheral device for testing. You can use any of the following peripheral samples:

* :ref:`peripheral_uart`
* :ref:`peripheral_hids_keyboard`
* :ref:`peripheral_hids_mouse`
* :ref:`peripheral_lbs`

> * :ref:`throughput`

Reviewer: May be `ble_throughput` instead of `throughput`? This could fix the doc build error.

Connected
Service discovery completed
Disconnect successful
SUCCESS
```

Testing Workflow
================

Manual Testing
--------------

1. Build and flash the application to a development kit
2. Power on a BLE peripheral device (for example, Nordic UART Service sample)
3. Monitor the console output for successful connection and service discovery
4. Verify that the application completes the full test cycle

Automated Testing
-----------------

The application is designed for integration with automated test frameworks:

* Supports CI/CD pipeline integration
* Provides structured test results
* Includes test procedure documentation
* Compatible with Nordic's test infrastructure

Troubleshooting
===============

Common Issues
-------------

1. **No devices discovered**: Ensure peripheral device is advertising with correct name
2. **Connection failures**: Check device compatibility and Bluetooth stack configuration
3. **Service discovery errors**: Verify peripheral device implements expected services
4. **Build errors**: Ensure all dependencies are properly installed

Debug Information
-----------------

Enable additional debug output by modifying `prj.conf`:

> .. code-block:: conf

Reviewer: May be "none" instead of "conf"?

Author: Section not needed, removed
